src/emit/polkadot/mod.rs

```diff
@@ -144,7 +146,16 @@ impl PolkadotTarget {
 
         emit_functions(&mut target, &mut binary, contract, ns);
 
-        let storage_initializer = emit_initializer(&mut target, &mut binary, contract, ns);
+        let function_name = CString::new(format!("{}:storage_initializer", contract.name)).unwrap();
```

Comment:
The rest of the functions use `::` as separators. In fact, the contract name is not necessary at all, I think.

Reply:
very nice!